### DISCUSSION TITLE
posts/use-asyncbarrier-to-sync-tasks/

### DISCUSSION THREAD

**Discussion by giscus** [BYJRK/byjrk.github.io#17] [Announcements]
# posts/use-asyncbarrier-to-sync-tasks/

如果在异步编程开发过程中遇到了多个异步任务需要在某一时刻全部完成后才能继续执行的情形，那么可以使用 AsyncBarrier 来帮助我们实现这一需求。这篇文章我们就来学习一下如何将它用在一个 WPF 项目中。

https://blog.coldwind.top/posts/use-asyncbarrier-to-sync-tasks/

**Comment by Nitload-NSI** [comment]
这个AsyncBarrier和.net中的Barrier类型有何异同，感觉在文章介绍地使用场景下，这两个类型似乎没有太大区别

**Reply by BYJRK** [reply]
区别就是标准库里面的 `Barrier` 只提供了同步方法；而 `AsyncBarrier` 类如其名，提供的是异步的方法

**Reply by Nitload-NSI** [reply]
> 区别就是标准库里面的 `Barrier` 只提供了同步方法；而 `AsyncBarrier` 类如其名，提供的是异步的方法

在自己写了一个WaitAsync并在业务中使用后才知道其具体作用。异步方法在等待时取消等功能上确实非常有用

**Comment by binglinwen** [comment]
如果其中一个线程在Barrier之前失败退出，是否导致另一个线程在Barrier卡住，这种情况如何让另一个线程继续执行
